### ISSUE TITLE
__cause__ is not followed

### ISSUE THREAD

**Issue by mohammadhzp** [alexmojaki/stack_data#39]
I am unable to view the correct cause of the error.
Maybe this is an issue with `utils.iter_stack`? I'm not sure.

> Sample code reproducing the issue:

```python

import sys

import stack_data


class MyException(Exception):
    def __init__(self, kind, *args):
        super().__init__(*args)

        self.kind = kind

    def __repr__(self):
        return f"Kind is {self.kind}"


def print_stack():  # Copied from README.md
    for frame_info in stack_data.FrameInfo.stack_data(sys.exc_info()[2]):
        if isinstance(frame_info, stack_data.FrameInfo):
            print(f"{frame_info.code.co_name} at line {frame_info.lineno}")
            print("-----------")
            for line in frame_info.lines:
                print(f"{'-->' if line.is_current else '   '} {line.lineno:4} | {line.render()}")

            for var in frame_info.variables:
                print(f"{var.name} = {repr(var.value)}")

            print()
        else:
            print(f"... {frame_info.description} ...\n")


def trouble():
    return 1 / 0


def process():
    try:
        return trouble()
    except Exception as e:
        raise MyException("ddd") from e


def main():
    try:
        process()
    except Exception:
        # import traceback
        # print(traceback.format_exc())
        print_stack()


main()


```

I can observe it using the built-in `traceback` module.

Python version: ` 3.10.6`

**Comment by alexmojaki** [comment]
This is normal, although it should definitely be documented better. `__cause__` and `__context__` are attributes of exception objects, not tracebacks themselves. Following those chains essentially means formatting multiple tracebacks.

There's a higher level API that I really need to document at some point that takes care of this so that you don't have to do so much manual formatting work. Here's an example:

```python
import stack_data

formatter = stack_data.Formatter()

try:
    1 / 0
except Exception as e:
    try:
        2 / 0
    except Exception as e2:
        formatter.print_exception(e2)
```

Output:

```
Traceback (most recent call last):
 File "....py", line 6, in <module>
       1 | import stack_data
       3 | formatter = stack_data.Formatter()
       5 | try:
-->    6 |     1 / 0
               ^^^^^
       7 | except Exception as e:
ZeroDivisionError: division by zero

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
 File "....py", line 9, in <module>
       6 |     1 / 0
       7 | except Exception as e:
       8 |     try:
-->    9 |         2 / 0
                   ^^^^^
      10 |     except Exception as e2:
ZeroDivisionError: division by zero
```

Here's the code that handles the chaining:

https://github.com/alexmojaki/stack_data/blob/0b69516005945eb99f443e2fb45b299a8031be23/stack_data/formatting.py#L98-L113

You'll have to look at the source code of that file to understand the formatting API, like I said I really should document it, but it shouldn't be hard to follow.

**Comment by mohammadhzp** [comment]
Thank you for your detailed response.
